client/client.go

```diff
@@ -442,9 +443,14 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 		return nil, nil, err
 	}
 
+	isWait := false
+	if req != nil && req.URL != nil {
+		isWait, _ = strconv.ParseBool(req.URL.Query().Get("wait"))
+	}
+
 	var hctx context.Context
 	var hcancel context.CancelFunc
-	if c.headerTimeout > 0 {
+	if !isWait && c.headerTimeout > 0 {
```

Comment:
@xiang90 if I understand your comment correctly, we can flush the buffer for each request like

```go
if f, ok := rw.(http.Flusher); ok {
    if f != nil {
      f.Flush()
    }
}
```

but this won't fix the root cause when there are more than 1 proxy?

If so, what do you suggest to handle such cases? I won't proceed with this PR
until we figure out the best approach.

Thanks for feedback!